src/schema-generation/billing-type-generator.ts

```diff
@@ -0,0 +1,157 @@
+import { GraphQLBoolean } from 'graphql';
+import memorize from 'memorize-decorator';
+import { RootEntityType } from '../model/implementation';
+import {
+    BILLING_KEYFIELD_NOT_FILLED_ERROR,
+    BinaryOperationQueryNode,
+    BinaryOperator,
+    ConfirmForBillingQueryNode,
+    EntitiesQueryNode,
+    ErrorIfNotTruthyResultValidator,
+    FieldQueryNode,
+    FirstOfListQueryNode,
+    LiteralQueryNode,
+    NOT_FOUND_ERROR,
+    PreExecQueryParms,
+    QueryNode,
+    RootEntityIDQueryNode,
+    TransformListQueryNode,
+    UpdateEntitiesQueryNode,
+    VariableQueryNode,
+    WithPreExecutionQueryNode
+} from '../query-tree';
+import { BILLING_MUTATION_INPUT_ARG } from '../schema/constants';
+import { getConfirmForBillingFieldName } from '../schema/names';
+import { OutputTypeGenerator } from './output-type-generator';
+import { QueryNodeField } from './query-node-object-type';
+
+export class BillingTypeGenerator {
+    constructor(readonly outputTypeGenerator: OutputTypeGenerator) {}
+
+    @memorize()
+    getMutationField(rootEntityType: RootEntityType): QueryNodeField | undefined {
+        if (!rootEntityType.billingEntityConfig || !rootEntityType.billingEntityConfig.billingKeyField) {
+            return undefined;
+        }
+        if (!rootEntityType.billingEntityConfig.billingKeyField.type.isScalarType) {
+            throw new Error('The BillingKeyField must be a scalar field.');
+        }
+        const inputType = rootEntityType.billingEntityConfig.billingKeyField.type.graphQLScalarType;
+        return {
+            name: getConfirmForBillingFieldName(rootEntityType.name),
+            type: GraphQLBoolean,
+            args: {
+                [BILLING_MUTATION_INPUT_ARG]: {
```

Comment:
something is off here, is this supposed to be the id or the key field? the argument is called id but its type is the key field type.

Reply:
I think it should be the ID right, because the key field will be fetched in the preexecs. The type then should just be `ID` I guess.